### PR TITLE
Equilateral triangle

### PR DESCRIPTION
**Pull request by CSwift8**
Created an exercise for user to determine third points of an equilateral triangle given the first two points.

### CODE REVIEW

File: exercises/practice/equilateral_triangle/.docs/instructions.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Instructions
+
+Use Z3 to implement a function that, when given two points on the 2D coordinate
```
Comment (Newby555) [inline exercises/practice/equilateral_triangle/.docs/instructions.md:3]: I don't think that you should say "Use Z3 to ..." since this will be under the Z3 track. Maybe just start the sentence with "Implement ..."

File: exercises/practice/equilateral_triangle/.docs/instructions.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# Instructions
+
+Use Z3 to implement a function that, when given two points on the 2D coordinate 
+plane, produces a third point such that the connection of these points
+results in an equilateral triangle.  However, two points should be output
+because there are two possible "third" points to produce an equilateral triangle.
```
Comment (Newby555) [inline exercises/practice/equilateral_triangle/.docs/instructions.md:6]: This sentence is kind of confusing. Maybe change to "The function will have two outputs because there are two possible "third" points to produce an equilateral triangle when given the first two points."

File: exercises/practice/equilateral_triangle/equilateral_triangle.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+from z3 import *
+
+def generate_third_points(first_point, second_point):
```
Comment (Newby555) [inline exercises/practice/equilateral_triangle/equilateral_triangle.py:3]: I would add a comment here indicating that this is where the user should add their code (you can use the same one that I have in mine to be consistent if you want).

File: exercises/practice/equilateral_triangle/equilateral_triangle.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    pass
+
+def convert_to_float(z3_number):
```
Comment (Newby555) [inline exercises/practice/equilateral_triangle/equilateral_triangle.py:6]: Maybe add a comment above this function indicating to the user that they shouldn't change it.

File: exercises/practice/equilateral_triangle/.meta/example.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
+from z3 import *
+
+def generate_third_points(first_point, second_point):
+    # Constants to determine x and y indices of input points
```
Comment (Newby555) [inline exercises/practice/equilateral_triangle/.meta/example.py:4]: Great idea!

File: exercises/practice/equilateral_triangle/equilateral_triangle_test.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+    def setUp(self):
+        # Display rational numbers as decimals instead of fractions
+        set_option(rational_to_decimal=True)
+
+        # Set Precision to 5 decimal places of outputted rational numbers
+        set_option(precision=EquilateralTriangleTest.NUM_DECIMAL_PLACES)
+
+    def test_same_y(self):
+        first_point = (0, 0)
```
Comment (grantg012) [inline exercises/practice/equilateral_triangle/equilateral_triangle_test.py:16]: This is a lot of repeated code. Might be easier to package most inputs and expect outputs into a list, write a check function, then go through everything in a for loop. See https://github.com/grantg012/exercism-z3/blob/goldbach/exercises/practice/goldbach/goldbach_test.py

File: exercises/practice/equilateral_triangle/.meta/example.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    return third_point_1, third_point_2
+
+def convert_to_float(z3_number):
+    """ 
+    Converts AlgebraicNumRef, IntNumRef, and RatNumRef
+    to a float from its string representation
+    """
+
+    # Get the Z3 Number's string representation
+    z3_number_string = str(z3_number)
```
Comment (grantg012) [inline exercises/practice/equilateral_triangle/.meta/example.py:73]: There are ways to avoid going to a string and back, though this code may be the most generic. https://github.com/grantg012/exercism-z3/blob/quadratic-roots/exercises/practice/quadratic-roots/quadratic_roots_test.py

Comment (CSwift8) [reply exercises/practice/equilateral_triangle/.meta/example.py:73]: Your implementation works as long as the input is of type 'RatNumRef.'  All the "numbers" that are passed into this (my) implementation are mostly of type 'AlgebraicNumRef,' which are defined as irrational numbers in Z3: [https://z3prover.github.io/api/html/classz3py_1_1_algebraic_num_ref.html](url).  Therefore, the RatNumRef.as_fraction() method could not be used

File: exercises/practice/equilateral_triangle/equilateral_triangle_test.py (hunk 6 above)
Comment (CSwift8) [reply exercises/practice/equilateral_triangle/equilateral_triangle_test.py:16]: I agree that there is a lot of repeated code and it should be condensed.  However, I do not think having one test case method is not such a good idea because if that test case method were to fail, you would not know which of the many inputs failed on that test case (unless there is some verbose detailing of the tests that I am unaware about).  I will soon implement a check function, and will combine the non-edge cases into one test case method.
